changelog_unreleased/json/18405.md

```diff
@@ -0,0 +1,48 @@
+#### Preserve number and string representation in `json-stringify` parser (#18405 by @fisker)
+
+Previously, we printed numbers and strings with `JSON.stringify()`, but it loses the original representation sometimes. Eg: super big or small numbers are wiped, special characters get unescaped. Technically, it doesn't change the value, but that's something the user expected and against [what we are concerned about](https://prettier.io/docs/rationale#strings). Prettier main keeps the original representation for them.
```

Comment:
```suggestion
Previous versions of Prettier used to print numbers and strings with `JSON.stringify()`. This lead to the loss of the original value representation in a few rare cases. For example, extremely big or small numbers were rounded and some special characters were unescaped. Technically, these transforms do not change how JSON values are read, but they are not something a formatter [is concerned about](https://prettier.io/docs/rationale#strings). Prettier main keeps the original representation, even if it can be simplified.
```